backend/src/main/java/com/festago/festival/repository/FestivalV1QueryDslRepository.java

```diff
@@ -0,0 +1,124 @@
+                    school.region
+                ),
+                festivalQueryInfo.artistInfo)
+            )
+            .from(festival)
+            .innerJoin(school).on(school.id.eq(festival.school.id))
+            .innerJoin(festivalQueryInfo).on(festivalQueryInfo.festivalId.eq(festival.id));
+    }
+
+    private BooleanExpression dynamicWhere(FestivalFilter filter, LocalDate currentTime, Long lastFestivalId,
+                                           LocalDate lastStartDate,
+                                           SchoolRegion region) {
+        if (hasCursor(lastStartDate, lastFestivalId)) {
+            return cursorBasedWhere(filter, currentTime, lastFestivalId, lastStartDate, region);
+        }
+
+        BooleanExpression filterResult = switch (filter) {
+            case PLANNED -> festival.startDate.gt(currentTime);
+            case PROGRESS -> festival.startDate.loe(currentTime).and(festival.endDate.goe(currentTime));
+            case END -> festival.endDate.lt(currentTime);
+        };
+        return addRegion(filterResult, region);
+    }
+
+    private boolean hasCursor(LocalDate lastStartDate, Long lastFestivalId) {
+        return lastStartDate != null && lastFestivalId != null;
+    }
+
+    private BooleanExpression cursorBasedWhere(FestivalFilter filter, LocalDate currentTime, Long lastFestivalId,
+                                               LocalDate lastStartDate, SchoolRegion region) {
+        BooleanExpression filterResult = switch (filter) {
+            case PLANNED -> festival.startDate.gt(lastStartDate)
+                .or(festival.startDate.eq(lastStartDate)
+                    .and(festival.id.gt(lastFestivalId)));
+            case PROGRESS -> festival.startDate.lt(lastStartDate)
+                .or(festival.startDate.eq(lastStartDate)
+                    .and(festival.id.gt(lastFestivalId)))
+                .and(festival.endDate.goe(currentTime));
+            case END -> festival.endDate.lt(currentTime);
+        };
+        return addRegion(filterResult, region);
+    }
+
+    private BooleanExpression addRegion(BooleanExpression filterResult, SchoolRegion region) {
+        if (region == null || region == SchoolRegion.기타) {
+            return filterResult;
+        }
+        return filterResult.and(school.region.eq(region));
+    }
+
+    private OrderSpecifier<LocalDate>[] dynamicOrderBy(FestivalFilter filter) {
+        return switch (filter) {
+            case PLANNED -> new OrderSpecifier[]{festival.startDate.asc(), festival.id.asc()};
+            case PROGRESS -> new OrderSpecifier[]{festival.startDate.desc(), festival.id.asc()};
+            case END -> new OrderSpecifier[]{festival.endDate.desc()};
+        };
+    }
+
+    private List<FestivalV1Response> removeNextPageData(List<FestivalV1Response> content, Pageable page) {
+        if (haveNextPageContent(content, page)) {
+            return content.subList(0, page.getPageSize());
+        }
```

Comment:
`sublist()`도 좋지만, `remove()` 메서드로 마지막 인덱스를 제거하는것은 어떤가요?
```suggestion
        if (haveNextPageContent(content, page)) {
            return content.remove(content.size() - 1);
        }
```